mouseion/mouseion/train_and_build_index.py

```diff
@@ -0,0 +1,82 @@
+import argparse
+import faiss
+import json
+import numpy
+import tqdm
+import pathlib
+
+def main(args):
+    """
+    Combines numpy embeddings from multiple npy files into a FAISS index.
+    """
+    try:
+        spec = json.load(args.spec.open("r"))
+    except Exception as e:
+        print(f"Error reading spec file: {e}")
+        return
+
+    total_embeddings = sum([numpy.load(shard["embeddings"], mmap_mode="r").shape[0] for shard in spec["shards"]])
+    is_trained = False
+
+    try:
+        first_shard = numpy.load(spec["shards"][0]["embeddings"], mmap_mode="r")
+    except Exception as e:
+        print(f"Error loading shard: {e}")
+        return
+
+    dim = first_shard.shape[1]
+    print(f'dimensionality is {dim}')
+
+    index = faiss.index_factory(dim, args.index_type, faiss.METRIC_L2)
+
+    if args.use_gpus:
```

Comment:
This high-level conditional could be factored out, so that we aren't mixing GPU-only paths with GPU-or-CPU paths.

Reply:
by factored out, do you mean placed in a separate function?